pagespeed/envoy/envoy_fetch.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include <vector>
+
+#include "envoy_url_async_fetcher.h"
+#include "net/instaweb/http/public/url_async_fetcher.h"
+#include "pagespeed/kernel/base/basictypes.h"
+#include "pagespeed/kernel/base/pool.h"
+
+#include "external/envoy_api/envoy/api/v2/core/http_uri.pb.h"
+#include "pagespeed/kernel/base/string.h"
+#include "pagespeed/kernel/base/thread_system.h"
+#include "pagespeed/kernel/http/response_headers.h"
+#include "pagespeed/kernel/http/response_headers_parser.h"
+#include "pagespeed_remote_data_fetcher.h"
+
+namespace net_instaweb {
+
+class EnvoyUrlAsyncFetcher;
+class EnvoyFetch;
+
+class PagespeedDataFetcherCallback : public PagespeedRemoteDataFetcherCallback {
```

Comment:
Please add some top level doc comments here (as well as for for the public methods).

Reply:
added doc comments as well as class description
change pushed in https://github.com/apache/incubator-pagespeed-mod/pull/1969/commits/8a92117ec74a99b4a1dffff479a71ca1b5db7717